### ISSUE TITLE
Found nil in applyPendingRelationships

### ISSUE THREAD

**Issue by omerbaskurt** [mentrena/SyncKit#78]
Hi @mentrena, I've been using SyncKit in my project for a few months. It's working pretty good for both private and shared databases. However, in rare cases, usually when constantly adding and deleting data and syncing in the process, I'm getting a crash in RealmSwiftAdapter.applyPendingRelationShips(). As far as I understand, it's a problem with RealmAdapter accessing a deleted realm object. This is the exact line that throws this error:
`for property in originObject!.objectSchema.properties { ... }`
Error source: originObject
Error: Thread 1: Fatal error: Unexpectedly found nil while unwrapping an Optional value
I followed the example on my implementation. What do you think could cause this?
Thanks

**Comment by mentrena** [comment]
Hi @omerbaskurt, does your app update/delete objects midway through the synchronisation process?
If so, I think I could do an easy fix to check if an object has been deleted before trying to apply any changes.

**Comment by omerbaskurt** [comment]
Hi, thanks for the reply. It's most likely what happens. If the user deletes an object, synchronization triggers. So, if the time between the deletion/update process is too short, it could happen in the midway of synchronization.
I want to ask you about something else, not related to this issue (I'll open a new issue if you want me to): 
I put a switch in my app to turn iCloud sync on/off. When it turns off, I delete record zones and call eraseLocalMetaData as follows
```
for adapter in _synchronizer.modelAdapters() { 
     _synchronizer.deleteRecordZone(for: adapter, withCompletion: nil)
}
_synchronizer.eraseLocalMetadata()
_synchronizer = nil
```
Initialize synchronizer when the switch turns on:
```
_synchronizer = QSCloudKitSynchronizer.cloudKitPrivateSynchronizer(containerName: CLOUDKIT_ZONE_NAME, configuration: RealmConfig.realmConfig)
```
When I turn it off and on again, in the very first synchronization I get a crash at RealmSwiftAdapter.serverChangeToken()
The reason of the crash is realmProvider is nil.
After that, it runs without an issue. Interestingly, if I force close the app and open it again after turning on the iCloud sync and not synchronizing afterward, it runs fine.
I hope I explained it clearly, thanks again.

**Comment by mentrena** [comment]
Just released 0.7.3 with a fix for this. Thanks!
